File: BackEnd/AspNetWebApi/AspNetWebApi/Models/Pedido.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.ComponentModel.DataAnnotations;

namespace AspNetWebApi.Models
{
    public class Pedido : BaseModelo
    {
        [Required]
        public Cliente Cliente { get; set; }

        public decimal ValorLiquido { get; set; }

        public decimal ValorBruto { get; set; }

        public decimal Desconto { get; set; }

        public DateTime DataHora { get; set; }

        [Required]
        public CondicaoPagamento CondicaoPagamento { get; set; }

        public DateTime CreatedAt { get; set; }

        public DateTime UpdatedAt { get; set; }

        public ICollection<PedidoItem> PedidoItems { get; set; }
    }
}

File: FrontEnd/Angular/src/app/classes/Pedido.ts
import { Produto } from "./Produto";
import { Condicaopagamento } from "./Condicaopagamento";
import { Cliente } from "./cliente";

export class PedidoItems {
  public Produto: Produto;
  public ProdutoId: number;
  public ValorUnitario?: number;
  public ValorBruto?: number;
  public ValorLiquido?: number;
  public Desconto?: number;
  public Quantidade?: number;
}

export class Pedido {
  public Id: number;
  public ClienteId: number;
  public Cliente: Cliente;
  public ValorLiquido: number;
  public ValorBruto: number;
  public Desconto: number;
  public DataHora: any;
  public CondicaoPagamentoId: number;
  public CondicaoPagamento: Condicaopagamento;
  public PedidoItems: PedidoItems[];
}
